### PR TITLE
Tag names of secondary variables better checked

### PR DESCRIPTION
**Pull request by chleh**
Thus it will be less likely to configure them wrong in custom processes.

The underlying "problem" is that the tag name has to be written twice in the source code in order to set up one secondary variable. Cf.
https://github.com/ufz/ogs/blob/master/ProcessLib/GroundwaterFlow/GroundwaterFlowProcess.h#L108
and
https://github.com/ufz/ogs/blob/master/ProcessLib/GroundwaterFlow/GroundwaterFlowProcess.h#L188

### CODE REVIEW

File: ProcessLib/SecondaryVariable.h [hunk 1]
```diff
@@ -115,6 +115,12 @@ class SecondaryVariableCollection final
 
         // read which variables are defined in the config
         for (auto const& tag_name : tag_names) {
+            if (!_all_secondary_variables.insert(tag_name).second)
+            {
+                ERR("Tag name <%s> has been specified twice as a secondary variable.");
```
Comment (norihiro-w) [inline ProcessLib/SecondaryVariable.h]: OGS_FATAL

File: ProcessLib/SecondaryVariable.h [hunk 2]
```diff
@@ -145,54 +151,76 @@ class SecondaryVariableCollection final
...
+                ERR("The secondary variable with name `%s' has already been "
+                    "set up.",
+                    var_name.c_str());
+                std::abort();
+            }
+        }
+        else if (_all_secondary_variables.find(tag_name) ==
+                 _all_secondary_variables.end())
+        {
+            ERR("The tag <%s> has not been registered to mark a secondary "
```
Comment (norihiro-w) [inline ProcessLib/SecondaryVariable.h]: OGS_FATAL

File: ProcessLib/SecondaryVariable.h [hunk 3]
```diff
@@ -145,54 +151,76 @@ class SecondaryVariableCollection final
...
-    {
-        return _map_tagname_to_varname[tag_name];
+            if (!_configured_secondary_variables
+                     .emplace(std::make_pair(
+                         var_name,
+                         SecondaryVariable<GlobalVector>{
+                             var_name, num_components, std::move(fcts)}))
+                     .second)
+            {
+                ERR("The secondary variable with name `%s' has already been "
```
Comment (norihiro-w) [inline ProcessLib/SecondaryVariable.h]: OGS_FATAL

**Comment by chleh** [comment]
Jenkins, test this please.

**Comment by norihiro-w** [comment]
:+1:

**Pull request merged**